main.ts

```diff
@@ -1,135 +1,110 @@
+// And gives you type-safety
+declare module "obsidian" {
+	interface Editor {
+		cm: {
+			state?: { wordAt: (offset: number) => EditorSelection };
+			findWordAt?: (pos: EditorPosition) => EditorSelection;
+		};
+	}
+}
 export default class SmarterMDhotkeys extends Plugin {
 	async onload() {
-		this.addCommand({
-			id: "smarter-asterisk-bold",
-			name: "Smarter Bold",
-			editorCallback: (editor: Editor, view: MarkdownView) => this
-				.expandAndWrap("**", "**", editor),
-		});
-		this.addCommand({
-			id: "smarter-underscore-bold",
-			name: "Smarter Underscore Bold",
-			editorCallback: (editor: Editor, view: MarkdownView) => this
-				.expandAndWrap("__", "__", editor),
-		});
-		this.addCommand({
-			id: "smarter-asterisk-italics",
-			name: "Smarter Italics",
-			editorCallback: (editor: Editor, view: MarkdownView) => this
-				.expandAndWrap("*", "*", editor),
-		});
-		this.addCommand({
-			id: "smarter-underscore-italics",
-			name: "Smarter Underscore Italics",
-			editorCallback: (editor: Editor, view: MarkdownView) => this
-				.expandAndWrap("_", "_", editor),
-		});
-		this.addCommand({
-			id: "smarter-comments",
-			name: "Smarter Comments",
-			editorCallback: (editor: Editor, view: MarkdownView) => this
-				.expandAndWrap("%%", "%%", editor),
-		});
-		this.addCommand({
-			id: "smarter-inline-code",
-			name: "Smarter Inline Code",
-			editorCallback: (editor: Editor, view: MarkdownView) => this
-				.expandAndWrap("`", "`", editor),
-		});
-		this.addCommand({
-			id: "smarter-highlight",
-			name: "Smarter Highlight",
-			editorCallback: (editor: Editor, view: MarkdownView) => this
-				.expandAndWrap("==", "==", editor),
-		});
-		this.addCommand({
-			id: "smarter-strikethrough",
-			name: "Smarter Strikethrough",
-			editorCallback: (editor: Editor, view: MarkdownView) => this
-				.expandAndWrap("~~", "~~", editor),
-		});
-		this.addCommand({
-			id: "smarter-wikilink",
-			name: "Smarter wikilink (internal link)",
-			editorCallback: (editor: Editor, view: MarkdownView) => this
-				.expandAndWrap("[[", "]]", editor),
-		});
-		this.addCommand({
-			id: "smarter-md-link",
-			name: "Smarter Markdown Link",
-			editorCallback: (editor: Editor, view: MarkdownView) => this
-				.expandAndWrap("[", "]()", editor),
+		COMMANDS.forEach((command) => {
+			const { id, name, before, after } = command;
+			this.addCommand({
+				id,
+				name,
+				editorCallback: (editor) =>
+					this.expandAndWrap(before, after, editor),
+			});
 		});
 
-		console.log ("Smarter MD Hotkeys loaded.");
+		console.log("Smarter MD Hotkeys loaded.");
 	}
 
 	async onunload() {
-		console.log ("Smarter MD Hotkeys unloaded.");
+		console.log("Smarter MD Hotkeys unloaded.");
 	}
 
 	expandAndWrap(beforeStr: string, afterStr: string, editor: Editor): void {
+		// Arrow functions are simpler
+		// No need to specify the return-type, Typescript can infer it
+		const offToPos = (offset: number) => editor.offsetToPos(offset);

```

Comment:
interesting... declaring a function with `const` gives me a bit of headaches though 😵‍💫

Reply:
Hahaha that's understable. I think it's useful to know that functions are objects like any other _thing_ in js. So a function can definitely be a const. It just happens to return something else

Reply:
Here's a good guide: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Functions/Arrow_functions